### PR TITLE
Fail a service's deployment if a future app event reveals that the service is in crashloop backoff

### PR DESCRIPTION
**Pull request by ferozemohideen**

### CODE REVIEW

**Review by jusrhee — APPROVED** [review]

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 1]
```diff
@@ -496,3 +513,71 @@ func getServiceNameFromPodName(podName, porterAppName string) string {
...
+	)
+
+	// convert the metadata to a map[string]interface{} because that is the type updateDeployEvent expects
+	// TODO: refactor updateDeployEvent so we don't have to do this
+	updateMetadataBytes, err := json.Marshal(updateMetadata)
+	if err != nil {
+		return telemetry.Error(ctx, span, err, "error marshaling update metadata")
+	}
+	updateMetadataMap := make(map[string]interface{})
+	json.Unmarshal(updateMetadataBytes, &updateMetadataMap)
```
Comment (d-g-town) [inline api/server/handlers/porter_app/create_and_update_events.go:578]: check err here

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 2]
```diff
@@ -496,3 +513,71 @@ func getServiceNameFromPodName(podName, porterAppName string) string {
...
+
+	updateMetadataMap["deploy_status"] = string(status)
+	_ = p.updateDeployEvent(ctx, porterAppName, porterAppId, deploymentTargetID, updateMetadataMap)
```
Comment (d-g-town) [inline api/server/handlers/porter_app/create_and_update_events.go:581]: why do we swallow these errors? If there is a reason, add it as a comment

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 3]
```diff
@@ -181,6 +181,23 @@ func (p *CreateUpdatePorterAppEventHandler) createNewAppEvent(ctx context.Contex
 				}
 			}
 		}
+
+		// before creating a new app_event type event, check the event for crashloop backoff
+		// if the event is a crashloop backoff, then update the service status of the deployment event associated it to FAILED, since the service's deployment will never succeed from crashloop backoff
+		// only applies to v2 apps (where the deployment target id is not empty)
+		if deploymentTargetID != "" {
+			updateMetadata, appEventFormattedCorrectly := appEventMatchesDetail(requestMetadata, "stuck in a restart loop")
```
Comment (d-g-town) [inline api/server/handlers/porter_app/create_and_update_events.go:189]: `const restartLoopSubstring := "stuck in a restart loop"` or something

Comment (d-g-town) [reply api/server/handlers/porter_app/create_and_update_events.go:189]: actually nah

**Review by d-g-town — APPROVED** [review]

**Pull request merged**